### PR TITLE
Telemetry Interfacing with Front End

### PR DESCRIPTION
**Pull request by JackCotter**
Implemented logic for [rust socket server](https://github.com/UVicRocketry/telemetry-information-server/tree/front-end-interface) to connect to the front end. Replaced socket.io implementation with native WebSockets. Also implemented the following:

 - SocketGateway immediately starts streaming mode, which prompts the server to send data every n seconds. Right now n is initiall 3 seconds.
 - Added setPacketFrequency context method that can be hooked into to change the packet frequency
 - Added setProtocol context method that can be hooked into to change from APRS to LORA or other.

**Important:** Use front-end-interface branch in rust server. That commit allows the client to toggle streaming mode.

### CODE REVIEW

**Comment by klemie** [comment]
Have you tested this with the server? For my LC changes you might need to change a bit more

**Comment by JackCotter** [comment]
@klemie I have not tested with server as it likely wont work. The rust implementation sends packets with a different structure to the front end and most likely they wont be able to be uploaded to the server without some changes. Should parsing that be included in this PR? If so, It would be helpful to get the telemetry receiver and decoder working so that we have some packets to test with.

File: client/src/utils/socket-context.tsx [hunk 1]
```diff
@@ -54,54 +45,59 @@ export const SocketGateway = ({ children }: PropsWithChildren<any>) => {
 	const [logs, setLogs] = useState<string[]>([]);
 	const [aprsPacket, aprsDispatch] = useReducer(aprsReducer, {} as IPacket);
 	const [loRaPacket, loraDispatch] = useReducer(loraReducer, {} as IPacket);
-	const socket = io('http://localhost:8086/toClient');
+	const [packetFrequency, setPacketFrequency] = useState<number>(3);
+	const [protocol, setProtocol] = useState<string>('APRS');
+
+	const port = 9193;
```
Comment (klemie) [inline client/src/utils/socket-context.tsx:51]: Magic number would be good to be be in the `.env` file

File: client/src/utils/socket-context.tsx [hunk 2]
```diff
@@ -54,54 +45,59 @@ export const SocketGateway = ({ children }: PropsWithChildren<any>) => {
 	const [logs, setLogs] = useState<string[]>([]);
 	const [aprsPacket, aprsDispatch] = useReducer(aprsReducer, {} as IPacket);
 	const [loRaPacket, loraDispatch] = useReducer(loraReducer, {} as IPacket);
-	const socket = io('http://localhost:8086/toClient');
+	const [packetFrequency, setPacketFrequency] = useState<number>(3);
+	const [protocol, setProtocol] = useState<string>('APRS');
```
Comment (klemie) [inline client/src/utils/socket-context.tsx:49]:
```suggestion
	const [packetFrequency, setPacketFrequency] = useState<number>(3);
	const [protocol, setProtocol] = useState<string>('APRS');
```
Why are these the default values?

File: client/src/utils/socket-context.tsx [hunk 3]
```diff
@@ -54,54 +45,59 @@ export const SocketGateway = ({ children }: PropsWithChildren<any>) => {
...
+
+	const port = 9193;
+	const ws = new WebSocket(`ws://localhost:${port}`);
+
+	const enableLiveMode = () => {
+		ws.send(JSON.stringify({ type: 'establish_stream', frequency: 2 }));
```
Comment (klemie) [inline client/src/utils/socket-context.tsx:55]: what does a frequency of 2 mean for this?

**Review by klemie — COMMENTED** [review]
Better late then never jackaboy. Looks pretty good just a couple hardcoded things I rather not have in main

File: client/src/utils/socket-context.tsx (hunk 1 above)
Comment (JackCotter) [reply client/src/utils/socket-context.tsx]: moved port to a .env .

File: client/src/utils/socket-context.tsx (hunk 2 above)
Comment (JackCotter) [reply client/src/utils/socket-context.tsx:49]: Protocol is either 'APRS' or 'LORA'. It is my understanding LORA is being phased out, so APRS is used here as the default. packetFrequency is the time between packets in streaming mode. So when it is set to a default of 2, that means there will be 2 seconds between packets. Both of these can be changed using socketContext hook

**Review by klemie — APPROVED** [review]
Works with docker. Merge time

**Pull request merged**